tools/docker/image-pull.sh

```diff
@@ -6,18 +6,18 @@
 # See LICENSE file for more details.
 ###############################################################
 
-scriptdir="$(cd "${0%/*}"; pwd)"
+scriptdir="$(cd "${0%/*}" || exit 1; pwd)"
```

Comment:
In other scripts we used this instead, it's more concise and the result is the same:

```suggestion
scriptdir="$(cd "${0%/*}" && pwd)"
```

Note that both solutions do silence the shellcheck issue, but do *not* exit the script if `cd` fails (but it's not something you have to fix IMO).

Reply:
If `scriptdir` is missing,  what is the point going on with script? It will just report missing functions of `functions.sh`.

Reply:
> If `scriptdir` is missing, what is the point going on with script?

Excellent argument to prefer `&&`! The problem with `exit 1` is that an unwary reader may think it will exit the script, while it actually just exists the subshell created by `$(...)`. To exit the script you have to do
```suggestion
scriptdir="$(cd "${0%/*}" && pwd)" || exit 1
```
or `set -e`.

Reply:
>The problem with exit 1 is that an unwary reader may think it will exit the script, while it actually just exists the subshell created by $(...)

Exactly. I now realize my comment was not clear enough: 
when I said "it's not something you have to fix IMO" what I meant was "you don't have to make the script exit on error" and **not** "you don't have to follow my suggestion at all".